.github/workflows/preview.yml

```diff
@@ -0,0 +1,45 @@
+name: Preview on Vercel
+
+on:
+  # Run this workflow whenever a pull request is opened, synchronized, or reopened
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+        # Checkout the repository
+      - name: Check Out Code
+        uses: actions/checkout@v2
+
+        # Deploy to Vercel
+      - name: Deploy to Vercel
+        env:
+          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+        run: |
+          npm install -g vercel
+          vercel deploy --target preview
+          export DEPLOYMENT_URL=$(vercel --json list | jq -r '.[0].url')
+
+        # Comment on the pull request with the deployment URL
+      - name: Comment on Pull Request
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Rename `GITHUB_TOKEN` with `GH_TOKEN`

Reply:
Also, as it is a `React` project, shouldn't you `build` it using `npm run build` before deploying tho? @sid86-dev

Reply:
> Also, as it is a `React` project, shouldn't you `build` it using `npm run build` before deploying tho? @sid86-dev

We use `vercel deploy` because it is more optimized for automated environments similar to `npm CI`, but underneath does the same work i,e; **build and deploy**. Further, it is the officially documented script for vercel remote deployment.